evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py

```diff
@@ -117,6 +117,7 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None) -> PolynomialDecomposer:
             period=self.seasonal_period,
         ).seasonal
         self.seasonality = self.seasonal[0 : self.seasonal_period]
+        self.trend = y - (y_detrended_with_time_index - self.seasonal) - self.seasonal
```

Comment:
is `(y_detrended_with_time_index - self.seasonal)` the constant?

Reply:
That value should be the residual!